### DISCUSSION TITLE
Group hover not work for nested components.

### DISCUSSION THREAD

**Discussion by muradsofi** [chakra-ui/chakra-ui#4674] [Ideas]
Let's say we have a nested component structure like that. In there if I hover the Link component, Button's Icon and Text component color also change but in the normal case, it must change when I hover on just Button. I think it's issue :/

```
<Link role="group"> 
 <Box _groupActive={{ color: "red" }}>
   <Button role="group">
       <Icon _groupHover={{ color: "red.500" }}/>
       <Text _groupHover={{ color: "red.300" }}> Test </Text>
   </Button>
 </Box>
</Link> 
```

**Comment by primos63** [comment]
TL;DR: It's not an issue. It could be a feature request, but if Tailwind hasn't implemented it, then I don't see Chakra doing so unless somebody wants to submit a PR. I just don't see it as a must-have.
___

I compared Chakra-UI, TailwindCSS and good ole HTML & CSS. Why Tailwind? Well while heavily inspired by Theme UI, Chakra also takes from Tailwind. Just do a search for `group hover` and you'll find entries pointing to Tailwind.

I have a [CodePen](https://codepen.io/primos63/pen/jOwmBWM) which shows what the expectation is, a [CodeSandbox](https://codesandbox.io/s/weathered-leftpad-s69ff) showing a variation of the example above and a [TailwindPlay](https://play.tailwindcss.com/AaAxDqgfcc) example.

The assumption is that one can nest groups, `role='group'`, and that subsequent `_group[pseudo class]` defined on an element will be associated with the parent having `role='group'`. That, however, is not the case. Both Chakra and Tailwind exhibit the same behavior. This is due to the way the CSS is generated:

```css
/* Box with _groupActive */
[role=group]:active .css-1offhpc {
  color: var(--chakra-colors-white);
}
/* Text with _groupHover within Button */
[role=group]:hover .css-liw04x {
  color: var(--chakra-colors-red-500);
}
/* Box with _groupActive within Button */
[role=group]:active .css-1etiy2h {
  color: var(--chakra-colors-pink-400);
}
```
How Chakra generates this can be found [here](https://github.com/chakra-ui/chakra-ui/blob/main/packages/styled-system/src/pseudos.ts). My verision above is an abbreviated one.

The `.css-` class names are the class names generated for the component. So what this says is to apply the pseudo element, `hover` or `active`,  to  the descendants of anything having the data attribute `role=group`. The first `role='group'` in the example will satisfy both the active and hover pseudo classes. If we remove the second `role='group'`, we see the same behavior. Therefore the second group is not necessary.

If we look at the HTML & CSS Codepen example, the CSS I wrote is
```css
.link[role='group']:active .group-active {
  color: white
}

.btn[role='group']:hover .group-hover {
  color: red;
}

.btn[role='group']:active .group-active {
  color: pink;
}
```
This provides the expectation I believe you are looking for. Here the class of the element with `role=group` which provides greater specificity. This isolates the hover and active pseudo classes to it's parent element having the attribute role=group. My classes are generic here. If this were from Chakra, then you would see the specific generated class name. This may be difficult to achieve. The class name of the element with role=group would have to be pushed onto a stack for use when a _group[pseudo class] is encountered and popped off once the block has been processed.

Tailwind's implementation operates the same as Chakra's. On Tailwind's github, there was a [feature request](https://github.com/tailwindlabs/tailwindcss/issues/1192) on Tailwind's Github, but it was closed by Adam in 2019. The most recent comment, posted 27 days ago, asked for a reconsideration of the feature. There has been no further activity. Nearly two years have passed. So how important is this? I don't see a significant portion of either userbase requesting the feature so at the moment I'd say not very.

**Reply by lt123345** [reply]
Tailwind supported group peers like `group/sidebar`, see https://github.com/tailwindlabs/tailwindcss/discussions/6034. Does chakra-ui want to support the same?
